### PR TITLE
User dashboard

### PR DESCRIPTION
**Pull request by bserge13**
## Description
UPDATE: changes to made to align with newly merged code. User dashboard/show spec now passing, but login is failing because of .join method usage. Unsure on how to approach that but will make members aware in slack. 

Completion of a users dashboard with link to edit, buttons to: log out, add pet, view pets, discover socials; all buttons also have the routes rolled (to be refactored). All testing passes. 

## Type of change
Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Describe testing implemented and what all it covers. 

- [ ]  fully tests happy and sad paths
- [x] happy path testing only

## Checklist

- [x] code has been self reviewed
- [ ] code runs without any errors
- [ ] all tests pass

### CODE REVIEW

File: app/views/users/show.html.erb
Comment (JustinSteel) [inline app/views/users/show.html.erb:1]: I think we should we put the log out button in the application view so it is at the top of all the pages.

**Review by ttakahashi1591 — COMMENTED** [review]
Thanks for getting this all built out and it all looks great. I did notice that coverage files are still being generated in this PR, but I did notice you have it removed on the other PR for the `Draper::Decorator` addition!

File: spec/features/users/show_spec.rb
Comment (JustinSteel) [inline spec/features/users/show_spec.rb:1]: looks good!

**Review by JustinSteel — APPROVED** [review]

File: app/views/users/show.html.erb
Comment (EricBelongea) [reply app/views/users/show.html.erb:1]: I'd agree with this. I think Having the Login and Log out is the application view and changing based on whether a user is logged in or not would be a great feature. As well as a dashboard button.

File: config/routes.rb [hunk 1]
```diff
@@ -8,9 +8,22 @@
...
+  # Socials 
+  get '/socials', to: 'socials#index', as: :socials
+
+  #User_Socials
+  get '/users/:id/socials', to: 'user_socials#index', as: :user_socials
+
+  # Dogs 
+  get '/users/:id/dogs/new', to: 'dogs#new', as: :add_dog
+  # post route for creating a users dog
+  get '/users/:id/dogs', to: 'dogs#index', as: :dogs
```
[the remark this answers is not in the document]
Comment (EricBelongea) [reply config/routes.rb:28]: Yes!

**Review by EricBelongea — CHANGES_REQUESTED** [review]

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class UsersController < ApplicationController
+  def show 
+    @user = UserFacade.get_by_id(params[:id])
```
[the remark this answers is not in the document]
Comment (bserge13) [reply app/controllers/users_controller.rb]: I don't think the current_user method is working as it should be. All errors for ID conversion when making a poro, array of all users being returned went away went away after reverting back to params[:id]. I might be missing something, but :id is what's being passed in the params within ApplicationController I saw as well

**Comment by EricBelongea** [comment]
Hey I took this on my local and seems like everything is working? I changed the cassettes and now I have all passing tests. I didn't have to change any code from @bserge13's last PR.

**Pull request merged**